crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -222,6 +338,81 @@ fn is_argument_only_used_in_recursion<'a>(
     true
 }
 
+fn is_property_only_used_in_recursion_jsx(
+    ident: &BindingIdentifier,
+    property_name: &str,
+    function_ident: &BindingIdentifier,
+    ctx: &LintContext,
+) -> bool {
+    let mut references = ctx
+        .semantic()
+        .symbol_references(ident.symbol_id.get().expect("`symbol_id` should be set"))
+        .peekable();
+
+    if references.peek().is_none() {
+        return false;
+    }
+
+    let function_symbol_id = function_ident.symbol_id.get().expect("`symbol_id` should be set");
+
+    for reference in references {
+        // Conditions:
+        // 1. The reference is inside a JSXExpressionContainer.
+        // 2. The JSXElement calls the recursive function itself.
+        // 3. The reference is in a JSXAttribute, and the attribute name has the same name as the function.
+        let Some(may_jsx_expr_container) = ctx.nodes().parent_node(reference.node_id()) else {
+            return false;
+        };
+        let AstKind::JSXExpressionContainer(_) = may_jsx_expr_container.kind() else {
+            // In this case, we simply ignore the references inside JSXExpressionContainer that are not single-node expression.
+            //   e.g. <Increment count={count+1} />
+            //
+            // To support this case, we need to check whether expression contains side-effect like ++val
+            return false;
+        };
+
+        let Some(attr) = ctx.nodes().ancestors(may_jsx_expr_container.id()).find_map(|node| {
+            if let AstKind::JSXAttributeItem(attr) = ctx.nodes().get_node(node).kind() {
```

Comment:
again, can use `iter_parents` to work with nodes directly (same goes for other usage in this method):

```suggestion
        let Some(attr) = ctx.nodes().iter_parents(may_jsx_expr_container.id()).find_map(|node| {
            if let AstKind::JSXAttributeItem(attr) = node.kind() {
```